plugins/BEdita/Core/src/Model/Behavior/GeometryBehavior.php

```diff
@@ -0,0 +1,184 @@
+
+        return new FunctionExpression(
+            $this->getConfig('distanceFunction'),
+            [
+                new FunctionExpression('ST_GeomFromText', $point1),
+                new FunctionExpression('ST_GeomFromText', $point2),
+            ],
+            [],
+            'float'
+        );
+    }
+
+    /**
+     * Parse coordinates.
+     *
+     * @param mixed $point Coordinates.
+     * @return float[]
+     * @throws \BEdita\Core\Exception\BadFilterException Throws an exception if value could not be parsed into coords.
+     */
+    public static function parseCoordinates($point)
+    {
+        if (empty($point)) {
+            throw new BadFilterException([
+                'title' => __d('bedita', 'Invalid data'),
+                'detail' => 'missing or empty coordinates',
+            ]);
+        }
+        if (!is_array($point)) {
+            $point = preg_split('/[\s,]/', (string)$point, 2);
+        }
+        $point = filter_var_array(array_values($point), FILTER_VALIDATE_FLOAT);
+        if (count($point) !== 2 || in_array(false, $point, true) || abs($point[0]) > 90 || abs($point[1]) > 180) {
+            throw new BadFilterException([
+                'title' => __d('bedita', 'Invalid data'),
+                'detail' => 'bad geo data format: ' . implode(' ', $point),
+            ]);
+        }
+
+        return $point;
+    }
+
+    /**
+     * Check if current DB supports geo operations.
+     *
+     * @return void
+     * @throws \BEdita\Core\Exception\BadFilterException Throws an exception if GIS SQL functions are not available.
+     */
+    public function checkGeoSupport()
+    {
+        $connection = $this->getTable()->getConnection();
+        if (!isset($this->hasGeoSupport)) {
+            try {
+                $query = new DatabaseQuery($connection);
+                $query = $query->select([
+                    'dist' => $this->getDistanceExpression([0, 0], [1, 1]),
+                ]);
+                $query->execute();
+
+                $this->hasGeoSupport = true;
+            } catch (\PDOException $e) {
+                $this->hasGeoSupport = false;
+            }
+        }
+
+        if ($this->hasGeoSupport !== true) {
+            throw new BadFilterException([
+                'title' => __d('bedita', 'Invalid data'),
+                'detail' => 'operation not supported on current database',
+            ]);
+        }
+    }
+
+    /**
+     * Find objects by geo coordinates.
+     *
+     * Create a query to filter objects using geo data: location objects are
+     * ordered by distance, from the nearest to the farthest using a center geo point.
+     *
+     * Accepted options are:
+     *   - 'center' with point coordinates, latitude and longitude
+     *   - 'from' with point coordinates, latitude and longitude
+     *   - 'radius' with positive decimal number
+     *
```

Comment:
center-radius-from pattern search meaning could be explained with more details
I'd explicitate options meanings in context, i.e.:
* Accepted options are:
*    - 'center' with point coordinates, latitude and longitude | center point for the search
*    - 'radius' with positive decimal number | radius to compute circle around the center, containing coordinates
*    - 'from' with point coordinates, latitude and longitude | another location point, used to compute distances from positions found

Maybe we could also indicate a dummy real example use:
"find all locations close to Trafalgare Square (center), within 2 km (radius), refering how far these location are from King Cross (from)